### ISSUE TITLE
 hwbase have no init

### ISSUE THREAD

**Issue by lljgithub** [RPi-Distro/raspi-gpio#10]
On debian, have no gpiomem so use /dev/mem and map use hwbase offset, but hwbase not inited.

**Comment by 6by9** [comment]
I've just observed this too. It's broken by 80fa7d0 Add support for 2711, but looks to be a relatively simple fix. (I've reverted that commit as I'm on a Pi3).

**Comment by 6by9** [comment]
At least for Pi3 it's only 1 line to reinstate
```
diff --git a/raspi-gpio.c b/raspi-gpio.c
index de5c0e4..392d473 100644
--- a/raspi-gpio.c
+++ b/raspi-gpio.c
@@ -759,6 +759,8 @@ int main(int argc, char *argv[])
             return 0;
         }
 
+       hwbase = get_hwbase();
+
         if (!hwbase)
             return 1;
 ```
Not tested on Pi4 yet.

**Comment by lurch** [comment]
Closed by #11 ?
